src/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -2151,40 +2133,36 @@ private String serializeMetadata(TownyObject obj) {
 
 	@Override
 	public boolean saveRegenList() {
-        
-        try (BufferedWriter fout = new BufferedWriter(new FileWriter(dataFolderPath + File.separator + "regen.txt"))) {
-            for (PlotBlockData plot : new ArrayList<>(TownyRegenAPI.getPlotChunks().values()))
-                fout.write(plot.getWorldName() + "," + plot.getX() + "," + plot.getZ() + newLine);
-            
-        } catch (Exception e) {
-            TownyMessaging.sendErrorMsg("Saving Error: Exception while saving regen file");
-            e.printStackTrace();
-            return false;
-            
-        }
+        queryQueue.add(() -> {
+			try (BufferedWriter fout = new BufferedWriter(new FileWriter(dataFolderPath + File.separator + "regen.txt"))) {
```

Comment:
Is there a reason that both saveRegen and saveSnapshot are manually writing to the file instead of using methods in `FileMgmt`?